config/initializers/session_limitable.rb

```diff
@@ -19,8 +19,10 @@
 
   if warden.authenticated?(scope) && options[:store] != false
     if record.unique_session_id != current_session_id
+      service_provider = warden.raw_session.dig('sp')
```

Comment:
This will maintain the SP for the user when theyre redirected back to the Error page.